pkg/pillar/types/global.go

```diff
@@ -114,6 +118,157 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota + 1
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+)
+
+type ConfigItemSpec struct {
+	Key      string
+	ItemType ConfigItemType
+
+	intMin     uint32
+	intMax     uint32
+	intDefault uint32
+
+	stringDefault   string
+	boolDefault     bool
+	triStateDefault TriState
+	agentSettings 	map[string]string
+}
+
+type ConfigItemSpecMap struct {
+	itemMap map[string]ConfigItemSpec
```

Comment:
We need to add agentSettings in this structure.

type ConfigItemSpecMap struct {
    itemMap map[string]ConfigItemSpec
    agentSettings 	map[string]ConfigItemSpec
}